python/docstring.cpp

```diff
@@ -0,0 +1,84 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2020 Scipp contributors (https://github.com/scipp)
+/// @file
+/// @author Neil Vaytet
+#include "docstring.h"
+
+Docstring &Docstring::description(const std::string &s, const bool append) {
+  if (append)
+    m_description += s;
+  else
+    m_description = s;
+  return *this;
```

Comment:
How about a private helper method taking a `string &` to a avoid the duplication, then this and the other similar methods below reduce to, e.g.,:
```cpp
return update(m_description, s, append);
```